src/devicemanager/devicemanager.cpp

```diff
@@ -9,6 +11,8 @@ PING_LOGGING_CATEGORY(DEVICEMANAGER, "ping.devicemanager");
 DeviceManager::DeviceManager()
     : _detector(new ProtocolDetector())
 {
+    QQmlEngine::setObjectOwnership(this, QQmlEngine::CppOwnership);
```

Comment:
can we get a comment? maybe it is common qt programming practice, but I don't know the reason for this addition.

Reply:
The idea here is to set the ownership of this class as cpp and not javascript, as documented here: https://doc.qt.io/qt-5/qqmlengine.html#ObjectOwnership-enum
The documentation is pretty explicit about what is going on here, we don't want this classes to be deleted by our QML engine but for the cpp scope.

Reply:
Thank you :+1: I did look on the qt doc, but did not find that enum on my own.